torch/csrc/api/include/torch/nn/options/transformer.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <torch/arg.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/types.h>
+#include <torch/enum.h>
+
+#include <torch/nn/modules/container/any.h>
+
+namespace torch {
+namespace nn {
+
+/// Options for the `Transformer` module
+///
+/// Example:
+/// ```
+///
+/// ```
+struct TORCH_API TransformerOptions {
+
+  using activation_t = c10::variant<enumtype::kReLU, enumtype::kGELU>;
+
+  TransformerOptions() = default;
+  TransformerOptions(int64_t d_model, int64_t nhead);
+  TransformerOptions(int64_t d_model, int64_t nhead, int64_t num_encoder_layers, int64_t num_decoder_layers);
```

Comment:
(question) Why did we choose to provide these two constructors and not, say,
```
TransformerOptions(int64_t d_model, int64_t nhead, int64_t num_encoder_layers, int64_t num_decoder_layers, dim_feedforward, dropout, activation)
```
?

Reply:
From the usage of python version, I found out that when people initialize the transformer, these two are mostly used. 
So I provide these two.

Reply:
Maybe add a comment to that effect in the code, so someone in the future doesn't go and try adding all of the overloads